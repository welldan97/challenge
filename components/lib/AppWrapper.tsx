import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode, Suspense } from 'react';
import ErrorBoundary from '../../lib/ErrorBoundary';

// SECTION: Main

interface Props {
  children: ReactNode;
}

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      suspense: true,
    },
  },
});

export default function AppWrapper({ children }: Props) {
  return (
    <ErrorBoundary fallback={<h1>Error</h1>}>
      <Suspense fallback={<h1>Loading...</h1>}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Suspense>
    </ErrorBoundary>
  );
}
