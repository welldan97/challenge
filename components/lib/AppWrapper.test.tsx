import { render } from '@testing-library/react';
import AppWrapper from './AppWrapper';

// SECTION: Main

describe('AppWrapper', () => {
  test('render', () => {
    const { container } = render(<AppWrapper>TEST</AppWrapper>);

    expect(container).toMatchSnapshot();
  });
});
