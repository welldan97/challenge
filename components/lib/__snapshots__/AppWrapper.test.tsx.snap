// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper render 1`] = `
<div>
  TEST
</div>
`;
